### ISSUE TITLE
bam2fq: Add -f option to enable separating paired-end reads

### ISSUE THREAD

**Issue by sjackman** [samtools/samtools#402]
Many tools cannot handle interleaved paired-end FASTQ files. It would be very helpful to separate the forward and reverse reads into two files. My suggestion for implementing this feature would be to add the `-f` option of `samtools view` to enable filtering out the paired-end forward and reverse reads into separate files.

**Comment by sjackman** [comment]
This shell snippet works:
```
samtools view -uf64 x.bam |samtools bam2fq - |gzip >x_1.fq.gz
samtools view -uf128 x.bam |samtools bam2fq - |gzip >x_2.fq.gz
```

**Comment by mp15** [comment]
This is possible on my bam2fq_improve branch feel free to test it out. I just need to make the main loop more readable.

> On 4 Jun 2015, at 00:06, Shaun Jackman <notifications@github.com> wrote:
> 
> Many tools cannot handle interleaved paired-end FASTQ files. It would be very helpful to separate the forward and reverse reads into two files. My suggestion for implementing this feature would be to add the -f option of samtools view to enable filtering out the paired-end forward and reverse reads into separate files.
> 
> —
> Reply to this email directly or view it on GitHub.
> 



-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Comment by sjackman** [comment]
Thanks, Martin.

**Comment by tseemann** [comment]
This is related to my request to Heng in seqtk: https://github.com/lh3/seqtk/issues/57

**Comment by jmarshall** [comment]
This is implemented in PR #263.
